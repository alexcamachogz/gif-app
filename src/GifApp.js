import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories)

  return (
    <>
      <div className="title-container">
        <h2 className="app-title">☀️ GIPHY
          API</h2>
        <p className="attribution">
          Powered by <a className="att-link"
                        href="https://giphy.com/"> Giphy</a>
        </p>
      </div>
      <AddCategory setCategories={setCategories}/>
      <ol>
        {categories.map(category => {
          return <GifGrid
            key={category}
            category={category}
          />
        })
        }
      </ol>
    </>
  )
}

export default GifApp