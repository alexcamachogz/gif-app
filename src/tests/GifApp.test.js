import { shallow } from 'enzyme'
import GifApp from '../GifApp'

describe('Testing <GifApp/>', () => {

  const wrapper = shallow(<GifApp/>)

  test('Should be show <GifApp/> properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should be show a category list', () => {
    const categories = ['One Piece', 'Harry Potter']
    const wrapper = shallow(<GifApp defaultCategories={categories}/>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})