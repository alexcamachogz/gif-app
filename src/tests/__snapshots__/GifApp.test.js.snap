// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifApp/> Should be show <GifApp/> properly 1`] = `
<Fragment>
  <div
    className="title-container"
  >
    <h2
      className="app-title"
    >
      ☀️ GIPHY API
    </h2>
    <p
      className="attribution"
    >
      Powered by 
      <a
        className="att-link"
        href="https://giphy.com/"
      >
         Giphy
      </a>
    </p>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Testing <GifApp/> Should be show a category list 1`] = `
<Fragment>
  <div
    className="title-container"
  >
    <h2
      className="app-title"
    >
      ☀️ GIPHY API
    </h2>
    <p
      className="attribution"
    >
      Powered by 
      <a
        className="att-link"
        href="https://giphy.com/"
      >
         Giphy
      </a>
    </p>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Harry Potter"
      key="Harry Potter"
    />
  </ol>
</Fragment>
`;
